pipelines.go

```diff
@@ -55,6 +56,60 @@ func (p *Pipeline) Lock() *Pipeline {
 	return p
 }
 
+func (p *Pipeline) ValidateRefIds() error {
+
+	var refidsSet = make(map[string]bool)
+	var requisitestagerefSet = make(map[string]bool)
+
+	//TODO: What should I return if stages do not exists? I think is valid to not have
+	//Check if there are stages, if not then do not process anything
+	if p.Stages != nil && len(p.Stages) > 0 {
+		//Iterate all stages
+		for _, stageMap := range p.Stages {
+
+			//ALL stages should have a redId
+			if _, exists := stageMap["refId"]; !exists {
```

Comment:
While true, I'm not sure if we want to pollute this validation method with knowledge about what's required and what isn't.

What do you think about adding a TODO here that links to a card in our JIRA to mention that this validation should live elsewhere? We can leave it in for now, and remove it later when we've matured the library a little bit.

Reply:
I will create the card and add the comment.

Reply:
Created: ENG-5293